sources/cql.y

```diff
@@ -848,17 +848,7 @@ math_expr[result]:
   ;
 
 expr[result]:
-  basic_expr  { $result = $basic_expr; }
-  | expr[lhs] '&' expr[rhs]  { $result = new_ast_bin_and($lhs, $rhs); }
```

Comment:
Yeah the problem here is that if you use math_expr for all of these then the LHS and RHS are limited to math_expr which means you cast parse something like `CAST(1 as REAL) + 1`

I'm adding new test cases to give a clean error on that.